inference-engine/tests/unit/cpu/mkldnn_memory_desc_test.cpp

```diff
@@ -45,314 +45,986 @@ TEST(MemDescTest, Conversion) {
+        hasLayoutTypeCpuBlkDesc(p.first.second, p.first.first, p.second);
+        hasLayoutTypeNcspNscp(p.first.second);
+    }
+}
 
-    memory::desc mkldnnDescTailC(dnnDims, dataType, memory::format_tag::acdb);
-    ASSERT_THAT(MKLDNNExtensionUtils::makeDescriptor(mkldnnDescTailC)->as<DnnlBlockedMemoryDesc>()->getOrder(), ElementsAre(0, 2, 3, 1));
+TEST(MemDescTest, blocksExtended) {
+    auto blocksExtended = [&] (const Shape& shape, const dnnl::memory::format_tag formatTag, const bool result) {
+        DnnlBlockedMemoryDescPtr dnnl_blk_desc_ptr = std::make_shared<DnnlBlockedMemoryDesc>(shape, mkldnn::memory::data_type::u8, formatTag);
+        CpuBlockedMemoryDescPtr cpu_blk_desc_ptr = std::make_shared<CpuBlockedMemoryDesc>(dnnl_blk_desc_ptr->getPrecision(), dnnl_blk_desc_ptr->getShape(),
+                                                                                          dnnl_blk_desc_ptr->getBlockDims(), dnnl_blk_desc_ptr->getOrder(),
+                                                                                          dnnl_blk_desc_ptr->getOffsetPadding(),
+                                                                                          dnnl_blk_desc_ptr->getOffsetPaddingToData(),
+                                                                                          dnnl_blk_desc_ptr->getStrides());
+
+        ASSERT_EQ(result, (*dnnl_blk_desc_ptr).blocksExtended());
+        ASSERT_EQ(result, (*cpu_blk_desc_ptr).blocksExtended());
+    };
 
-    memory::desc mkldnnDescBlockedC(dnnDims, dataType, memory::format_tag::aBcd16b);
-    ASSERT_THAT(MKLDNNExtensionUtils::makeDescriptor(mkldnnDescBlockedC)->as<DnnlBlockedMemoryDesc>()->getOrder(), ElementsAre(0, 1, 2, 3, 1));
+    std::pair<std::pair<dnnl::memory::format_tag, Shape>, bool> payload[] {
+            {{ dnnl::memory::format_tag::nChw16c,     Shape(SizeVector{1, 16, 10, 10}) }, false},
+            {{ dnnl::memory::format_tag::nChw16c,     Shape(SizeVector{1, 1, 10, 10}) }, true},
+            {{ dnnl::memory::format_tag::nhwc,        Shape(SizeVector{4, 2, 10, 7 }) }, false},
+            {{ dnnl::memory::format_tag::nchw,        Shape(SizeVector{4, 2, 10, 7 }) }, false},
+            {{ dnnl::memory::format_tag::nChw8c,      Shape(SizeVector{1, 1, 10, 10}) }, true},
+            {{ dnnl::memory::format_tag::nChw16c,     Shape(ngraph::PartialShape{{16}, {16}, {-1, -1}, {-1, -1}}) }, false},
+            {{ dnnl::memory::format_tag::nChw16c,     Shape(ngraph::PartialShape{{16}, {6}, {-1, -1}, {-1, -1}}) }, true},
+            {{ dnnl::memory::format_tag::nhwc,        Shape(ngraph::PartialShape{{8, 16}, {16}, {8, 24}, {10, 20}}) }, false},
+            {{ dnnl::memory::format_tag::nchw,        Shape(ngraph::PartialShape{{8, 16}, {8, 16}, {8, 24}, {10, 20}}) }, false},
+            {{ dnnl::memory::format_tag::nChw8c,      Shape(ngraph::PartialShape{{-1, -1}, {-1, -1}, {-1, -1}, {-1, -1}}) }, false},
```

Comment:
looks like we need throw exception if blocked dims == UNDEF
i.e. nChw8c channel == UNDEF other dims maybe defined